helper/mfa/duo/duo.go

```diff
@@ -0,0 +1,135 @@
+package duo
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/duosecurity/duo_api_golang/authapi"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func DuoPaths() []*framework.Path {
+	return []*framework.Path{
+		pathDuoConfig(),
+		pathDuoAccess(),
+	}
+}
+
+func DuoPathsSpecial() []string {
+	return []string {
+		"duo/access",
+		"duo/config",
+	}
+}
+
+func DuoHandler(req *logical.Request, d *framework.FieldData, resp *logical.Response) (
+	*logical.Response, error) {
+	duoConfig, err := GetDuoConfig(req)
+	if err != nil || duoConfig == nil {
+		return logical.ErrorResponse("Could not load Duo configuration"), nil
+	}
+
+	duoAuthClient, err := GetDuoAuthClient(req, duoConfig)
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	username, ok := resp.Auth.Metadata["username"]
```

Comment:
This is a really critical piece that we should document. The mapping between Duo username and Vault username needs to be well understood so it can be setup. Maybe just add some notes to the docs.